### DISCUSSION TITLE
Sonoff THR316D - switching relays based on temperature.

### DISCUSSION THREAD

**Discussion by JanPichl** [arendst/Tasmota#19600] [Q&A] [answered]
Hi
I have successfully uploaded the Tasmota firmware to the Sonoff THR316D.  Temperature measurement and manual control of both built-in relays works fine. I don't know how to make the relay switch on automatically when the temperature drops below 50°C and switch off automatically when it reaches 60°C. The only thing I could find in the web interface is the time schedule. Please is it possible to set the automatic switching/unswitching of the built-in relays based on the measured temperature?

Thanks

Sorry for the pictures, the installed firmware is in CZ

Main screen
![obrazek](https://github.com/arendst/Tasmota/assets/26009131/616c5d03-d705-4ac7-ba44-9dde4fcbca68)

Settings screen
![obrazek](https://github.com/arendst/Tasmota/assets/26009131/76637180-d3a3-446d-8bf1-f1165dd2c4fc)

Time schedule screen
![obrazek](https://github.com/arendst/Tasmota/assets/26009131/104bca4f-2049-40a7-a81c-cdaf4192e2b0)

**Comment by lalo-uy** [comment, marked as answer]
In the rules documentation ther is a thermostat example.

https://tasmota.github.io/docs/Rules/#simple-thermostat-example

**Comment by jlazkano** [comment]
Hello,

I have same device, could you show me how to install the Tasmota firmware? Need I solder any PINs? Or just upload a file in a web interface?

Regards.

**Reply by videobuff** [reply]
I have the same problem. Connected an USB to RS232, then started various tasmota uploaders.  They all stop becuase there is an issue wth a header. I pressed the reset button while powering up, but still no luck. Anyone else ?
